builtin/rebase.c

```diff
@@ -723,6 +723,16 @@ int cmd_rebase(int argc, const char **argv, const char *prefix)
 		int flag;
 
 		if (!force_rebase) {
+			/* Lazily switch to the target branch if needed... */
+			if (switch_to) {
+				strbuf_reset(&buf);
+				strbuf_addf("rebase: checkout %s", switch_to);
+				if (reset_hard(switch_to)) {
+					ret = !!error(_("could not switch to "
+							"%s"), switch_to);
+					goto cleanup;
+				}
```

Comment:
I just realized that this is missing a crucial bit: while it updates the `HEAD` to `switch_to`, it fails to update `HEAD` to the symbolic ref indicated by `switch_to`. That is a problem, and we have discussed this before (keyword: `REF_NO_DEREF`, and: `detach_head_to()` vs `reset_hard()`), but we failed to fix it so far.

In particular, I think this will completely replace your 3/4 of the currently-active patch series that you sent to the mailing list, and it will need to be fixed before you send another iteration.